### PR TITLE
feat: Adding Post Connection Timeout + Write Interval Timeouts

### PR DESCRIPTION
**Pull request by davidterpay**

### CODE REVIEW

File: oracle/config/websocket.go [hunk 1]
```diff
@@ -70,6 +78,10 @@ type WebSocketConfig struct {
 	// to the websocket endpoint.
 	ReconnectionTimeout time.Duration `json:"reconnectionTimeout"`
 
+	// PostConnectionTimeout is the timeout for the provider to wait after a connection
+	// is established before sending messages.
+	PostConnectionTimeout time.Duration `json:"postConnectionTimeout"`
```
Comment (g3ndo9kar9) [inline oracle/config/websocket.go:83]: Timeout is a misleading name here? More like Wait, or smth

File: providers/base/websocket/handlers/ws_query_handler.go [hunk 2]
```diff
@@ -159,6 +159,9 @@ func (h *WebSocketQueryHandlerImpl[K, V]) start() error {
 		return errors.ErrDialWithErr(err)
 	}
 
+	// Wait for the connection timeout before sending the initial payload(s).
+	time.Sleep(h.config.PostConnectionTimeout)
```
Comment (g3ndo9kar9) [inline providers/base/websocket/handlers/ws_query_handler.go:163]: we can get rate-limited after the first connection?

**Review by g3ndo9kar9 — APPROVED** [review]
lgtm

File: providers/base/websocket/handlers/ws_query_handler.go (hunk 2 above)
Comment (davidterpay) [reply providers/base/websocket/handlers/ws_query_handler.go:163]: no messages should have been sent to the connection so should not be possible on init. it is possible if the connection is multi-plexed but as long as the multi-plexing is spaced out sufficiently we shouldnt see this issue. Ideally the time between spinning up concurrent connections is greater than the rate limit configs.

**Pull request merged**